server/src/auth-router.ts

```diff
@@ -114,6 +118,14 @@ router.get(
       }
       response.redirect('/profile/settings?success=' + success.toString());
     } else {
+      // pass the challenge_team to frontend
+      user.challenge_team = challenge_team;
```

Comment:
I'm confused about what's happening here. I'm missing context, but `user` looks like a temporary object?

Reply:
This is INDEED quite confusing! But the `user `belongs to the `request `object. 
The authentication server (or SSO server, be it auth0 or mozzila SSO) embedded the user information (email, verified or not, etc.) inside the `request `object, and send it back to `/callback`.
With the redirection below, the user object is passed to the frontend. That's how the email shows up in the `Profile `page.